experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,523 @@
+
+    mat = zero_matrix(R, n, n)
+    for j in 1:n
+        f = images_of_b[j]
+        x = mixed_ring()
+        # express f as a linear combination of degree_basis
+        for i in 1:n
+            c = coeff(f, leading_exponent(b[i]))
+            mat[i,j] = c / leading_coefficient(b[i])
+        end
+    end
+    return mat
+end
+
+#computes symmetric degree basis (of the first m variables) WITH multinomial coefficients!
+function degree_basis(R::MPolyRing, t::Int)
+    m = ngens(R)
+    C = zero_matrix(ZZ, m, m)
+    for i in 1:m
+      C[i,i] = -1 
+    end
+    d = zeros(m)
+    A = ones(m)
+    b = [t]
+    P = polyhedron((C, d), (A, b))
+    L = lattice_points(P)
+    W = Vector{MPolyRingElem}(undef,0)
+    for l in L
+        v = R(1)
+        for i in 1:m
+            v = v*gen(R,i)^l[i]
+        end
+        v = v*multinomial(t,l)
+        push!(W,v)
+    end
+    #we reverse here to get the natural order of a degree basis, eg x^2, 2xy, y^2.
+    return reverse(W)
+end
+
+#used to compute multinomial expansion coefficients (used in degree_basis)
+function multinomial(n::Int, v::Union{Vector{Int64},PointVector{ZZRingElem}})
+    l = length(v)
+    x = 1
+    for i in 1:l
+        x = x*factorial(v[i])
+    end
+    return Int(factorial(n)/x)
+end
+
+##########################
+#Invariant Rings of Reductive groups
+##########################
+mutable struct RedGrpInvRing
+    field::Field
+    poly_ring::MPolyDecRing #graded
+    
+    group::ReductiveGroup
+    representation::RepresentationReductiveGroup
+    
+    fundamental::Vector{MPolyDecRingElem}
+    primary::Vector{MPolyDecRingElem}
+    secondary::Vector{MPolyDecRingElem}
+    
+    reynolds_operator::Function
+    
+    NullConeIdeal::MPolyIdeal
+    
+    #Invariant ring of reductive group G (in representation R), no other input.
+    function RedGrpInvRing(R::RepresentationReductiveGroup) #here G already contains information n and rep_mat
+        z = new()
+        n = ncols(R.rep_mat)
+        z.representation = R
+        z.group = R.group
+        G = z.group
+        z.field = G.field
+        z.poly_ring, __ = graded_polynomial_ring(G.field, "X" => 1:n)
+        z.reynolds_operator = reynolds_v_slm
+        return z
+    end
+    
+    #to compute invariant ring ring^G where G is the reductive group of R. 
+    function RedGrpInvRing(R::RepresentationReductiveGroup, ring::MPolyDecRing)
+        n = ncols(R.rep_mat)
+        n == ngens(ring) || error("The given polynomial ring is not compatible.")
+        z = new()
+        if isdefined(R, :weights)
+            #dosomething
+        end
+        z.representation = R
+        z.group = R.group
+        G = R.group
+        z.field = G.field
+        z.poly_ring = ring
+        z.reynolds_operator = reynolds_v_slm
+        return z
+    end
+end
+
+invariant_ring(R::RepresentationReductiveGroup) = RedGrpInvRing(R)
+invariant_ring(ring::MPolyDecRing, R::RepresentationReductiveGroup) = RedGrpInvRing(R, ring)
```

Comment:
Would be nice to also have `invariant_ring(G::ReductiveGroup)` which then automatically uses the canonical/natural representation (for convenience)